### PR TITLE
Add impersonation allowed

### PR DESCRIPTION
**Pull request by Brett55**
Users can now set impersonation_allowed to True at the task level

order = gbdx.Task('Auto_Ordering', cat_id=catalog_ids)
order.impersonation_allowed = True

### CODE REVIEW

**Comment by nricklin** [comment]
unit tests are failing

File: gbdxtools/simpleworkflows.py [hunk 1]
```diff
@@ -228,6 +228,7 @@ def __init__(self, __interface, __task_type, **kwargs):
         self.inputs = Inputs(self.input_ports, task=self)
         self.outputs = Outputs(self.output_ports, self.name)
         self.batch_values = None
+        self.impersonation_allowed = None
```
Comment (dmitryzv) [inline gbdxtools/simpleworkflows.py]: Prepend with _

**Comment by dmitryzv** [comment]
Please add unit tests for the new functionality.

**Comment by Brett55** [comment]
Fixed bug, added 2 unit tests

**Pull request merged**